doc/en/components/grids/_shared/state-persistence.md

```diff
@@ -93,39 +108,141 @@ const gridState: IGridState = state.getState(false);
+```
+
+```typescript
+var gridState = document.getElementById('gridState') as IgcGridStateComponent;
+
+// get an `IgcState` object, containing all features original state objects, as returned by the grid public API
+const state: IgcState = gridState.extractState();
+
+// get all features` state in a serialized JSON string
+const stateString: string = gridState.extractStateAsString();
+
+// get the sorting and filtering expressions
+const sortingFilteringStates: IgcState = gridState.extractState(['sorting', 'filtering']);
+```
+<!-- end: WebComponents -->
+
+```tsx
+<IgrGrid>
+    <IgrGridState ref={(ref) => { gridState = ref; }}></IgrGridState>
+</IgrGrid>
+```
+
+```tsx
+// get an `IgrState` object, containing all features original state objects, as returned by the grid public API
+const state: IgrState = gridState.extractState([]);
+
+// get all features` state in a serialized JSON string
+const stateString: string = gridState.extractStateAsString([]);
+
+// get the sorting and filtering expressions
+const sortingFilteringStates: IgrState = gridState.extractState(['sorting', 'filtering']);
+```
 
 ```razor
-GetState blazor snippet
+<{ComponentSelector}>
+    <IgbGridState @ref="gridState"></IgbGridState>
+</{ComponentSelector}>
+
+@code {
+    // get an `IgbState` object, containing all features original state objects, as returned by the grid public API
+    State state = gridState.ExtractState(new string[0]);
+
+    // get all features` state in a serialized JSON string
+    string stateString = gridState.ExtractStateAsString(new string[0]);
+
+    // get the sorting and filtering expressions
+    State sortingFilteringStates = gridState.ExtractState(new string[] { "sorting", "filtering" });
+}
 ```
 
+<!-- Angular -->
 `SetState` - The `SetState` method accepts the serialized JSON string or `IGridState` object as argument and will restore the state of each feature found in the object/JSON string.
+<!-- end: Angular -->
 
+<!-- Blazor, React, WebComponents -->
+`ApplyState` - The method accepts a `State` object as argument and will restore the state of each feature found in the object or specified features as second argument.
+
+`ApplyStateFromString` - The method accepts a serialized JSON string as argument and will restore the state of each feature found in the JSON string or specified features as second argument.
+<!-- end: Blazor, React, WebComponents -->
+
+<!-- Angular -->
 ```typescript
 state.setState(gridState);
 state.setState(sortingFilteringStates)
 ```
+<!-- end: Angular -->
+
+<!-- WebComponents -->
+```typescript
+gridState.applyState(gridState);
+gridState.applyStateFromString(gridStateString);
+gridState.applyState(sortingFilteringStates)
+```
+<!-- end: WebComponents -->
+
+```tsx
+gridState.applyState(gridState, []);
+gridState.applyStateFromString(gridStateString, []);
+gridState.applyState(sortingFilteringStates, [])
+```
 
 ```razor
-SetState blazor snippet
+gridState.ApplyState(gridState, new string[0]);
```

Comment:
I wonder if it would be better to force the features to be optional, since it would look cleaner. I'll see if there's an easy fix we can add for optional array handling.